### PR TITLE
upgraded to three 0.103.

### PR DESCRIPTION
**Pull request by makimenko**
- switched to standard three/examples/jsm
- removed ColladaLoader (no standard jsm yet)
- removed js/EnableThreeJs

### CODE REVIEW

File: src/app/three-examples/objects/loaders/obj-loader.directive.ts [hunk 1]
```diff
@@ -3,9 +3,9 @@ import * as THREE from 'three';
 import { AbstractObject3D } from '../abstract-object-3d';
 import { AbstractModelLoader } from './abstract-model-loader';
 
-import '../../js/EnableThreeExamples';
-import 'three/examples/js/loaders/OBJLoader';
-import 'three/examples/js/loaders/MTLLoader';
+import * as THREE_OBJ from 'three/examples/jsm/loaders/OBJLoader';
```
Comment (ComFreek) [inline src/app/three-examples/objects/loaders/obj-loader.directive.ts]: Couldn't we just do `import {OBJLoader} from 'three/...` and `import {MTLLoader} from 'three/...` one line below? That would seem cleaner to me.

File: src/app/three-examples/objects/loaders/obj-loader.directive.ts [hunk 2]
```diff
@@ -42,9 +42,10 @@ export class ObjLoaderDirective extends AbstractModelLoader {
         if (this.texturePath !== undefined) {
           this.mtlLoader.setTexturePath(this.texturePath);
         }
-        this.mtlLoader.load(this.material, material => {
-          material.preload();
-          this.loader.setMaterials(material);
+        this.mtlLoader.load(this.material, materialCreator => {
+          materialCreator.preload();          
+          var x:any = materialCreator;// seems wrong type in js module ts. Hack (as any)
+          this.loader.setMaterials(x);
```
Comment (ComFreek) [inline src/app/three-examples/objects/loaders/obj-loader.directive.ts]: Easier: `this.loader.setMaterials(materialCreator as any)`

File: src/app/three-examples/objects/loaders/obj-loader.directive.ts [hunk 3]
```diff
@@ -42,9 +42,10 @@ export class ObjLoaderDirective extends AbstractModelLoader {
...
           this.mtlLoader.setTexturePath(this.texturePath);
         }
-        this.mtlLoader.load(this.material, material => {
-          material.preload();
-          this.loader.setMaterials(material);
+        this.mtlLoader.load(this.material, materialCreator => {
+          materialCreator.preload();          
+          var x:any = materialCreator;// seems wrong type in js module ts. Hack (as any)
+          this.loader.setMaterials(x); 
           this.loader.load(this.model, model => {
```
Comment (ComFreek) [inline src/app/three-examples/objects/loaders/obj-loader.directive.ts]: Also easier: `this.loader.load(this.model, resolve)`

File: src/app/three-examples/controls/orbit-controls.directive.ts [hunk 4]
```diff
@@ -10,8 +10,7 @@ import {
   QueryList
 } from '@angular/core';
 import * as THREE from 'three';
-import '../js/EnableThreeExamples';
-import 'three/examples/js/controls/OrbitControls';
+import * as Orbit from 'three/examples/jsm/controls/OrbitControls';
```
Comment (ComFreek) [inline src/app/three-examples/controls/orbit-controls.directive.ts]: `import {OrbitControls} from 'three/...'`

**Comment by ComFreek** [comment]
Looks good to me apart from some minor changes we could make to make the code smaller and easier to read.

File: src/app/three-examples/objects/loaders/obj-loader.directive.ts (hunk 2 above)
Comment (ComFreek) [reply src/app/three-examples/objects/loaders/obj-loader.directive.ts]: Btw, I found [`MaterialCreator.getAsArray()`](https://github.com/mrdoob/three.js/blob/dev/examples/jsm/loaders/MTLLoader.d.ts#L99). Have you tried using that one?
I also wonder why the cast you're doing worked (it did, right?). I couldn't find any prototype/array indexing magic in Three.js' source code ([MTLLoader.js](https://github.com/mrdoob/three.js/blob/dev/examples/jsm/loaders/MTLLoader.js)). So actually, it shouldn't work.

Comment (makimenko) [reply src/app/three-examples/objects/loaders/obj-loader.directive.ts]: > Btw, I found MaterialCreator.getAsArray(). Have you tried using that one?

Unfortunately, it's compiling, but failing at runtime.
```
ERROR TypeError: this.materials.create is not a function
    at OBJLoader.parse (OBJLoader.js:702)
    at Object.onLoad (OBJLoader.js:402)
```

**Comment by makimenko** [comment]
> Looks good to me apart from some minor changes we could make to make the code smaller and easier to read.

Thanks for your comments!

File: src/app/three-examples/objects/loaders/obj-loader.directive.ts (hunk 2 above)
Comment (ComFreek) [reply src/app/three-examples/objects/loaders/obj-loader.directive.ts]: @makimenko Uh, sounds like a bug then. I could file it this evening, I guess, if you push your current code.

**Pull request merged**